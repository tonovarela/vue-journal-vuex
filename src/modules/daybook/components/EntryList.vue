<template>
<div class="entry-list-container">
    <div class="px-x2 pt-2 ">
        <input type="text"  v-model="term" class="form-control" placeholder="Buscar entrada" />
    </div>

<div class="mt-2 d-flex flex-column" >
    <button class="btn btn-primary mx-3" @click="$router.push({name:'entry',params:{id:'new'}})" >
        <i class="fa fa-plus-circle"></i>  
        Nueva entrada
    </button>

</div>

    <div class="entry-scollarea">
        <Entry v-for="entry in  entriesByTerm"  :entry="entry" :key="entry.id"></Entry>
    </div>
   </div>
</template>

<script>
import {  defineAsyncComponent } from 'vue'
import  { mapGetters} from 'vuex'
export default {
    components:{
       Entry: defineAsyncComponent(()=>import("./Entry")) 
    },
    computed: {
        ...mapGetters('journal',['getEntriesByTerm']),
          entriesByTerm(){              
           return this.getEntriesByTerm(this.term)
        }
    },
    data() {
        return {
            term:''
        }
    }

}
</script>

<style lang="scss" scoped>
.entry-list-container {
    border-right: 1px solid #2c3e50;
    height: calc(100vh - 20px);
}

.entry-scrollarea {
   // height: calc(100vh - 110px);
    overflow: scroll;
}
</style>
