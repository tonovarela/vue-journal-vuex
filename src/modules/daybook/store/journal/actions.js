import journalApi from '@/api/journalApi'


export const loadEntries = async({commit})=>{
      const  {data}=  await journalApi.get('entries.json');
      if (!data){
        commit('setEntries', []);      
        return ;  
      }

      const entries = [];
      for(let id   of Object.keys(data)){
          entries.push( {id, ...data[id]});          
      }
      //console.log(entries);
      commit('setEntries', entries);      

}

export const updateEntry = async({commit},entry)=>{
    const { date, picture,text} = entry;
    const dataToSave = { date, picture,text};
    await journalApi.put(`entries/${entry.id}.json`, dataToSave);
    dataToSave.id =entry.id;
    commit('updateEntry',{...dataToSave});
}

export const createEntry = async({commit},entry)=>{
    const { date, picture,text} = entry;
    const dataToSave = { date, picture,text};
    const e =await journalApi.post(`entries.json`, dataToSave);
    const id =e.data.name;
    entry.id=id;
    commit('addEntry',{...entry});
}

export const deleteEntry = async({commit},id)=> {        
    await journalApi.delete(`entries/${id}.json`);
    commit('deleteEntry',id);
    
}