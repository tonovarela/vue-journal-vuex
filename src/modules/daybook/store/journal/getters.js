// export const myGetter = (state)=>{
    //return state

// }

export const getEntriesByTerm = ( /*state*/ )=>{
//    return state

}


export const getEntriesById = ( /*state*/ )=>{
    //return state

}