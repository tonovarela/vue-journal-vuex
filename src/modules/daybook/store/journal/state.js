export default ()=> ({
    isLoading:true,
    entries:[
      //   {
      //      //id:new Date().getTime(),
      //      date:new Date().toDateString(),
      //      id:'1',
      //       text:"Lorem, ipsum dolor sit amet consectetur adipisicing elit. Doloribus modi consequuntur ab blanditiis cumque ducimus illum totam at inventore laudantium sed, provident quisquam expedita a repellat praesentium ipsam velit optio!",
      //       picture:null
      //    },
      //    {
      //       //id:new Date().getTime()+1000,
      //       date:new Date().toDateString(),
      //       id:'2',
      //       text:"sds",
      //       picture:null
      //    },
      //    {
      //       //id:new Date().getTime()+2001,
      //       date:new Date().toDateString(),
      //       id:'3',
      //       text:"afff",
      //       picture:null
      //    },
      //    {
      //       //id:new Date().getTime()+3002,
      //       date:new Date().toDateString(),
      //       id:'4',
      //       text:"xx",
      //       picture:null
      //    }
    ]

})