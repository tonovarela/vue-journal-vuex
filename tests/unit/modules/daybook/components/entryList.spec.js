
import { createStore } from 'vuex';

import { shallowMount } from '@vue/test-utils'
import EntryList from '@/modules/daybook/components/EntryList.vue'
import journal from '@/modules/daybook/store/journal'
import { journalState } from '../../../mock-data/test-journal-state';



describe('Pruebas en el Entry List', () => {

    const createVuexStore = (inicialState) => createStore({
        modules: {
            journal: {
                ...journal,
                state: { ...inicialState }
            }
        }
    })

    let wrapper;
    let mockRouter;

    beforeEach(() => {
        const store = createVuexStore(journalState);
         mockRouter = {
            push: jest.fn()
        }

        wrapper = shallowMount(EntryList, {
            global: {
                mocks: {
                    $router: mockRouter
                },
                plugins: [store]
            }
        })

    })


    test('Debe de llamar el getEntriesByTerm sin termino y mostrar 2 entradas', () => {
        expect(wrapper.findAll('entry-stub').length).toBe(3);
        expect(wrapper.html()).toMatchSnapshot();
    })

    test('Debe de llamar el getEntriesByTerm y filtrar las entradas', async() => {
        const input = wrapper.find('input')
        await input.setValue('Entrada');
        expect(wrapper.findAll('entry-stub').length).toBe(2);

    })
    test('El boton de nuevo debe de direccionar a new',()=>{
        wrapper.find("button").trigger('click')
        expect(mockRouter.push).toHaveBeenCalledWith({
            name:'entry',params:{id:'new'}
        })
    })
})