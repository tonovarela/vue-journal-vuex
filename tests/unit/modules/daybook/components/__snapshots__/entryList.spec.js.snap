// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Entry List Debe de llamar el getEntriesByTerm sin termino y mostrar 2 entradas 1`] = `
<div class="entry-list-container">
  <div class="px-x2 pt-2"><input type="text" class="form-control" placeholder="Buscar entrada"></div>
  <div class="mt-2 d-flex flex-column"><button class="btn btn-primary mx-3"><i class="fa fa-plus-circle"></i> Nueva entrada </button></div>
  <div class="entry-scollarea">
    <entry-stub entry="[object Object]"></entry-stub>
    <entry-stub entry="[object Object]"></entry-stub>
    <entry-stub entry="[object Object]"></entry-stub>
  </div>
</div>
`;
